File: cpp/1695.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> seen;
        int res = 0;
        int cur = 0;
        int l = 0;


        for (int r = 0; r < nums.size(); r++) {
            if (seen.find(nums[r]) != seen.end()) {
                while (nums[l] != nums[r]) {
                    seen.erase(nums[l]);
                    cur -= nums[l];
                    l++;
                }
                cur -= nums[l];
                l++;
            }
            cur += nums[r];
            seen.insert(nums[r]);
            res = max(cur, res);
        }
        return res;
    }
};

File: py/1695.py
class Solution:
    def maximumUniqueSubarray(self, nums: list[int]) -> int:
        seen = set()
        l = 0
        max_sum = 0
        current_sum = 0


        for r in range(len(nums)):
            if nums[r] in seen:
                while nums[l] != nums[r]:
                    seen.remove(nums[l])
                    current_sum -= nums[l]
                    l += 1
                current_sum -= nums[l]
                l += 1
            seen.add(nums[r])
            current_sum += nums[r]
            max_sum = max(max_sum, current_sum)
        return max_sum